src/main/java/seedu/address/model/appointment/Appointment.java

```diff
@@ -0,0 +1,161 @@
+package seedu.address.model.appointment;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Hashtable;
+import java.util.Objects;
+import java.util.Random;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.doctor.Doctor;
+import seedu.address.model.patient.Patient;
+import seedu.address.model.shared.Date;
+
+/**
+ * Represents an Appointment in the contacts book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Appointment {
+
+    // Hashtable of ID => appointment
+    private static Hashtable<Integer, Appointment> appointmentById = new Hashtable<>();
+
+    // ID of the appointment
+    private final Integer id;
+
+    // Data fields
+    private final Doctor doctor;
+    private final Patient patient;
+    private final Date date;
+    private final Time time;
+
+    /**
+     * Constructor with all fields.
+     * Every field must be present and not null.
+     */
+    public Appointment(Integer id, Doctor doctor, Patient patient, Date date, Time time) {
+        requireAllNonNull(id, doctor, patient, date, time);
+
+        this.id = id;
+        this.doctor = doctor;
+        this.patient = patient;
+        this.date = date;
+        this.time = time;
+
+        doctor.addAppointment(this);
+        patient.addAppointment(this);
+
+        appointmentById.put(id, this);
+    }
+
+    /**
+     * Constructor with only data fields.
+     * Automatically assigns a new random unique ID.
+     */
+    public Appointment(Doctor doctor, Patient patient, Date date, Time time) {
+        this(getNewId(), doctor, patient, date, time);
+    }
+
+    private static Integer getNewId() {
```

Comment:
I like this way of generating unique ids. Though, it might generate negative integers. (Is this something we should worry about?)